dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -102,19 +102,24 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
                                        TensorShape<1>{300*1024});
       hw_decoder_images_staging_.Resize(shapes, TypeInfo::Create<uint8_t>());
 #if defined(NVJPEG_PREALLOCATE_API)
-      // TODO(awolant): How to expose chroma subsampling to the user?
-      if (spec.GetArgument<int>("preallocate_width_hint") > 0 &&
-          spec.GetArgument<int>("preallocate_height_hint") > 0) {
-        LOG_LINE << "Using NVJPEG_PREALLOCATE_API" << std::endl;
-        NVJPEG_CALL(nvjpegDecodeBatchedPreAllocate(
-          handle_,
-          state_hw_batched_,
-          CalcHwDecoderBatchSize(hw_decoder_load_, max_batch_size_),
-          spec.GetArgument<int>("preallocate_width_hint"),
-          spec.GetArgument<int>("preallocate_height_hint"),
-          NVJPEG_CSS_444,
-          NVJPEG_OUTPUT_RGB));
-      }
+      // call nvjpegDecodeBatchedPreAllocate to use memory pool for HW decoder even if hint is 0
+      auto preallocate_width_hint = spec.GetArgument<int>("preallocate_width_hint");
+      auto preallocate_height_hint = spec.GetArgument<int>("preallocate_height_hint");
+      DALI_ENFORCE(preallocate_width_hint >= 0 && preallocate_width_hint >= 0,
+                   make_string("Provided preallocate_width_hint=", preallocate_width_hint, ", and ",
+                   "preallocate_height_hint=", preallocate_height_hint, " should not be ",
+                   "negative."));
+      preallocate_width_hint = preallocate_width_hint < 0 ? 0 : preallocate_width_hint;
```

Comment:
Nitpick: after the enforce, the values are guaranteed to be >= 0, so no need for this ternary.

Reply:
Leftover. Fixed.